### DISCUSSION TITLE
config option so `x` only deletes files from dir where last selection was made

### DISCUSSION THREAD

**Discussion by brett-mahar** [jarun/nnn#1880] [Ideas]
A few times, I have selected a few files to move, gone into the new directory, got distracted when I noticed stuff that needed deleting, selected that stuff, pressed `x`, and so accidently deleted the stuff I wanted to copy from the other directory. 

Is there some way I can config my nnn to not delete from more than one directory at the same time? I never want to do this on purpose.

If there is no such option, I think it would be quite useful to have. Or even to change current behaviour to not allowing deleting from more than one dir at a time, just display an "error: cannot delete from more than one dir at a time" or whatever.

Its quite easy to delete a huge directory you were just trying to move, and not even realise till a long time later :-o

**Comment by darukutsu** [comment]
when you make selection you can delete(or other operations which supports this) for hovered file or selection(bottom left corner).
![image](https://github.com/jarun/nnn/assets/54768193/90222612-785a-45d4-8299-b5f69bbdebe7)
If you want to make multiple selections and delete selected from dirA while preserving dirB, you could make script to parse [`.selection`](https://github.com/jarun/nnn/wiki/concepts#selection) and rm -rf things from dirA bind it to `;x` or whatever.

**Reply by brett-mahar** [reply]
Hey Daru,

Yes I did mean when doing multiple selections!

Anyway thanks to your advice I have written a plugin and bound to ;x via the NNN_PLUG='x:nnn-delete-pwd-only' env variable

Here's the script in case anyone's brain is like mine and forgets there is selected stuff in another folder:

#!/usr/bin/dash

cat .config/nnn/.selection | perl -pe 's/\0/\n/g' > /tmp/.selectedList
lastDir=$(cat /tmp/.selectedList | tail -n 1 | xargs dirname)                                                                                                                
cat /tmp/.selectedList | grep $lastDir | xargs rm -rf
    
Cheers!

**Comment by N-R-K** [comment]
> Is there some way I can config my nnn to not delete from more than one directory at the same time?

No. But you can use `$NNN_TRASH` + `trash-cli` to make it so that accidental deletes can be reverted back easily.

**Reply by brett-mahar** [reply]
Oh thanks @N-R-K that's a good idea. I will use that in conjunction with my plugin script.

**Comment by brett-mahar** [comment]
So I rewrote the "delete only from the last selected directory" plugin in Julia (less annoying to debug than the shell script version I posted earlier) and made it more rubust (so it can handle spaces, |, and other special characters in the filename, and not freak out if selction file is empty):

```
#!/usr/bin/env julia

raw = ""::String

function cleanup()
  sleep(2)
  run(`clear`)
  exit()
end

if isfile("/home/brett/.config/nnn/.selection") == false
  println("selection file does not exist")
  cleanup()
end

f = open("/home/brett/.config/nnn/.selection", "r")
  global raw = read(f, String)       
close(f)

splitArray = split(raw, "\u0")

if isempty(splitArray[1])
  println("nothing selected")
  cleanup()
end

lastDir = dirname(splitArray[length(splitArray)])

for i in 1:length(splitArray)
  if lastDir == dirname(splitArray[i])
    hasToGo = splitArray[i]
    println(" removing:", hasToGo)
    strrr = escape_string(hasToGo)
    rm(strrr, force=true, recursive=true)
  end
end

rm("/home/brett/.config/nnn/.selection", force=true)
cleanup()
```
